src/android/PurchasesPlugin.java

```diff
@@ -463,6 +469,28 @@ private void beginRefundRequestForProductId(CallbackContext callbackContext) {
         callbackContext.success();
     }
 
+    @PluginAction(thread = ExecutionThread.UI, actionName = "showInAppMessages")
+    private void showInAppMessages(JSONArray messageTypes, CallbackContext callbackContext) {
+        if (messageTypes == null) {
+            CommonKt.showInAppMessagesIfNeeded(this.cordova.getActivity());
+        } else {
+            ArrayList<InAppMessageType> messageTypesList = new ArrayList<>();
+            for (int i = 0; i < messageTypes.length(); i++) {
+                try {
+                    int messageTypeInt = messageTypes.getInt(i);
+                    InAppMessageType messageType = InAppMessageType.values()[messageTypeInt];
```

Comment:
This can throw an `ArrayIndexOutOfBoundsException` in Android because it only has one enum entry whereas we have 3 in the hybrids. We should probably have added some kind of `InAppMessageType.fromInt(x)` method in Android/PHC to make this easier.

Reply:
Good catch... yeah I fixed in https://github.com/RevenueCat/cordova-plugin-purchases/pull/430/commits/dc519787f60527244e1870ce9d09124237e5e893

Though it would probably be a good idea to have some common logic in Android/PHC.

Reply:
We can address moving the common logic to PHC/Android after this release